### PR TITLE
Eliminate shared this from std/internal/math/biguintcore.d

### PR DESCRIPTION
**Pull request by andralex**
cc @don-clugston-sociomantic @donc @ibuclaw

### CODE REVIEW

File: std/internal/math/biguintcore.d [hunk 1]
```diff
@@ -45,20 +45,19 @@ else
...
-
 private:
 // Limits for when to switch between algorithms.
-immutable size_t CACHELIMIT;   // Half the size of the data cache.
+// Half the size of the data cache.
+@nogc nothrow pure size_t getCacheLimit()
+{
+    import core.cpuid;
+    return (cast(size_t function() @nogc nothrow pure)
+        (() => core.cpuid.datacache[0].size * 1024 / 2))();
```
Comment (wilzbach) [inline std/internal/math/biguintcore.d]: I'm taking a bold guess here that this won't have a big performance-wise impact (at least with LDC).

File: std/internal/math/biguintcore.d [hunk 2]
```diff
@@ -45,20 +45,19 @@ else
...
-    CACHELIMIT = core.cpuid.datacache[0].size*1024/2;
-}
-
 private:
 // Limits for when to switch between algorithms.
-immutable size_t CACHELIMIT;   // Half the size of the data cache.
+// Half the size of the data cache.
+@nogc nothrow pure size_t getCacheLimit()
+{
+    import core.cpuid;
```
Comment (wilzbach) [inline std/internal/math/biguintcore.d]: Why no selective imports?

File: std/internal/math/biguintcore.d (hunk 1 above)
Comment (wilzbach) [inline std/internal/math/biguintcore.d]: Btw

>  `datacase` is scheduled for deprecation. Please use `dataCaches` instead.

File: std/internal/math/biguintcore.d (hunk 2 above)
Comment (andralex) [inline std/internal/math/biguintcore.d]: Not an established style (especially after we got rid of the hijacking), but I'm happy to add it.

File: std/internal/math/biguintcore.d (hunk 1 above)
Comment (andralex) [inline std/internal/math/biguintcore.d]: Yah, a performance hit is unlikely to be even measurable. Will change s/datacache/dataCaches/

**Comment by MetaLang** [comment]
@andralex CircleCI errors are fixed. Still good to go on this?

**Pull request merged**